app/src/forms/rbac/routes.js

```diff
@@ -15,39 +14,39 @@ routes.get('/idps', async (req, res, next) => {
   await controller.getIdentityProviders(req, res, next);
 });
 
-routes.get('/forms', keycloak.protect(`${config.get('server.keycloak.clientId')}:admin`), async (req, res, next) => {
+routes.get('/forms', async (req, res, next) => {
```

Comment:
For getUserForms/getFormUsers and setUserForms/setFormUsers, I think you will want to do a check (in middleware possibly, or controller or service) that the current user has team management permissions on any form id. Most importantly on the set* calls.  Don't want someone mucking with forms they don't have permission on.

Reply:
Hmm damn, yeah. For the set ones I think we can alter the userAccess.hasFormPermissions check to look for a query param (if no path param) and use that middleware as it's used elsewhere.
For getUserForms not really sure.. that one is based on just userID isn't it. So no permission check would work.

Reply:
probably put it in the service, filter out form ids user doesn't have access to before acting on them.  don't think need to raise 4** errors, just remove them from the work and the response should indicate what was actually done.

Reply:
Ok @parc-jason does this change make sense to you (on this PR already and I've tested all cases out)?
- Change the usedAccess.hasFormPermissions to check a query param if it's optionally invoked on a path with a formId query param (line 48 of that file)

In Rbac routes
- getFormUsers
- setFormUsers
- setUserForms
All these operate on a specific formID, and only on that form. So invoke the new hasFormPermissions check with the query param.

For
- getUserForms
This one is really an administrative function, it returns all form for a user that the caller specifies. (the FE should be using /current). So secure with admin flag.
This is actually not called by the FE right now always so doesn't matter. But would be useful for an admin panel.
Will rethink some of these routes in context of an admin panel in the future (likely duplicate with different permissions under a /admin or something)

Reply:
yeah, i think that's the best initial go.  you need to make sure the UX is calling setUserForms with a formId parameter (because I don't believe it is mandatory, but is needed for hasFormPermissions.

Reply:
Great, thanks for the review and catching this!